app/services/UserService.scala

```diff
@@ -0,0 +1,46 @@
+package services
+
+import com.google.inject.{Inject, Singleton}
+import common.User
+import dao.UserDao
+import exceptions.{InvalidDataException, YaltaBaseException}
+import play.api.Logging
+
+import scala.concurrent.{ExecutionContext, Future}
+import misc.validator.UserExtentionValidator._
+import misc.validator.{Validated, ValidationFailed}
+
+@Singleton
+class UserService @Inject()(userDao: UserDao)(implicit ec: ExecutionContext) extends Logging {
+
+  def get(id: Long): Future[Option[User]] = userDao.getUser(id)
+
+  def get(name: String): Future[Option[User]] = userDao.getUser(name)
+
+  def createUser(user: common.User): Future[common.User] = {
+    user.validation match {
+      case Validated => userDao.create(user)
+      case ValidationFailed(reason) => throw new InvalidDataException(reason)
+    }
+  }
+
+  def createUser(name: String, pass: String, role: common.Role): Future[common.User] =
+    createUser(new common.User(null, name, pass, role))
+
+  def update(name: String, change: common.User => common.User): Future[User] = {
+    userDao.getUser(name).map {
+      case None => throw new YaltaBaseException(s"User $name does not exist, can't update")
+      case Some(user) => user
+    }.flatMap {
+      user => userDao.update(change(user))
+    }
+  }
+
+  def changeRole(name: String, newRole: common.Role): Future[common.User] = {
```

Comment:
R we doin this????? Why??

Reply:
Why not?

Reply:
I strongly believe that admin should be admin forever
And driver should forever be driver
If person wants to be both - let's create two different accounts
Same with role migration

Reply:
Hmmm, let's leave it like that for now and just not call this from API, k?

Reply:
Meh
k